### DISCUSSION TITLE
bind with custom python ML model

### DISCUSSION THREAD

**Discussion by thapecroth** [postgresml/postgresml#509] [Q&A] [answered]
Hello,

I love what you guys have done. I am wondering whether it's possible to bind my own custom model built in PyTorch. I am looking to make a recommendation system. If so, is there any guide or API template that I should follow? Thanks!

**Comment by montanalow** [comment, marked as answer]
If you’re able to publish your PyTorch model via huggingface, you can load it in v1: https://postgresml.org/user_guides/transformers/pre_trained_models/

We’re still working to port the huggingface APIs to rust in v2 (PRs welcome :)

The issue is that models require some metadata etc. We’re looking at some different ways to incorporate deep learning APIs ergonomically in v2, which are more complex than other more traditional algorithms. We’ll get there sooner or later.
